table/tables/tables.go

```diff
@@ -323,6 +336,74 @@ func (t *Table) rebuildIndices(ctx context.Context, h int64, touched []bool, old
 	return nil
 }
 
+// check unique indices constrains
+func (t *Table) checkCreateUnqIdxAvail(txn kv.Transaction, r []interface{}, h int64) (duplicate bool, recordID int64, err error) {
```

Comment:
you can return kv.ErrKeyExists directly for duplicated and use it for outer check.